GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -632,155 +633,205 @@ private void writeAnswer(
+                        .size()));
                 break;
 
-            // Two different formats:
-            // old: Foo|Bar|Baz (never contains code)
-            // new: [{"name": "Foo", "code": "A"}, {"name": "Bar", "code": "B"} ... ]
-            case CASCADE:
-                List<Map<String, String>> cascadeNodes = new ArrayList<>();
-
-                if (value.startsWith("[")) {
-                    try {
-                        cascadeNodes = OBJECT_MAPPER.readValue(value,
-                                new TypeReference<List<Map<String, String>>>() {
-                                });
-                    } catch (IOException e) {
-                        log.warn("Unable to parse CASCADE response - " + value, e);
-                    }
-                } else if (!value.isEmpty()) {
-                    for (String name : value.split("\\|")) {
-                        Map<String, String> m = new HashMap<>();
-                        m.put("name", name);
-                        cascadeNodes.add(m);
-                    }
-                }
+            case OPTION:
+                cells.add(optionCellValue(value));
+                break;
 
-                boolean allCodesEqualsName = true;
-                for (Map<String, String> cascadeNode : cascadeNodes) {
-                    String code = cascadeNode.get("code");
-                    String name = cascadeNode.get("name");
+            default:
+                cells.add(sanitize(value));
+                break;
+        }
 
-                    if (code != null && name != null
-                            && !code.toLowerCase().equals(name.toLowerCase())) {
-                        allCodesEqualsName = false;
-                        break;
-                    }
-                }
-                if (allCodesEqualsName) {
-                    for (Map<String, String> cascadeNode : cascadeNodes) {
-                        cascadeNode.put("code", null);
-                    }
-                }
+        int col = startColumn;
+        for (String cellValue : cells) {
+            if (questionType == QuestionType.NUMBER) {
+                createCell(row, col, cellValue, null, Cell.CELL_TYPE_NUMERIC);
+            } else {
+                createCell(row, col, cellValue);
+            }
+            col++; // also takes care of padding in case no cell content added
+        }
+    }
 
-                if (useQuestionId) {
-                    // +------------+------------+-----
-                    // |code1:value1|code2:value2| ...
-                    // +------------+------------+-----
+    private static String dateCellValue(String value, Long questionId) {
+        try {
+            return ExportImportUtils.formatDate(ExportImportUtils.parseDate(value));
 
-                    int levelCount = questionDto.getLevelNames().size();
-                    int padCount = levelCount - cascadeNodes.size();
+        } catch (Exception e) {
```

Comment:
Is this exception ever thrown? Seems you already designed all the date parsing functions to already catch and handle the exceptions.  If there are no exceptions then maybe we can also leave out the `questionId` parameter?